crates/shared/src/maintenance.rs

```diff
@@ -24,50 +25,11 @@ impl ServiceMaintenance {
             metrics: Metrics::instance(observe::metrics::get_storage_registry()).unwrap(),
         }
     }
-}
 
-#[cfg_attr(test, mockall::automock)]
```

Comment:
Just moved down a bit to collapse 2 `impl ServerMaintenance` blocks into 1 to avoid confusion.